gnovm/pkg/gnolang/machine.go

```diff
@@ -1934,9 +1934,11 @@ func (m *Machine) PopAsPointer(lx Expr) PointerValue {
 		return ptr
 	case *CompositeLitExpr: // for *RefExpr
 		tv := *m.PopValue()
+		hv := m.Alloc.NewHeapItem(tv)
 		return PointerValue{
-			TV:   &tv, // heap alloc
-			Base: nil,
+			TV:    &hv.Value,
+			Base:  hv,
+			Index: 0,
```

Comment:
```suggestion
```

Reply:
I think 0 is nice... HIV are expected to have value 0, we could maybe extend it to allow a limited number in the future, say for grouping 1-refcount objects together with their parent.

Reply:
I only suggested removing it because the default value for this type is 0, so no need to set it unless it is being set to a non-default value.